store/src/sql/entities/stream.rs

```diff
@@ -0,0 +1,61 @@
+use cid::Cid;
+use sqlx::{sqlite::SqliteRow, Row};
+
+use super::StreamCid;
+
+#[derive(Debug, Clone, sqlx::FromRow)]
+pub struct StreamRow {
+    pub cid: Vec<u8>,
+    pub sep: String,
+    pub sep_val: Vec<u8>,
+}
+
+impl StreamRow {
+    pub fn insert() -> &'static str {
+        "INSERT INTO ceramic_one_stream (cid, sep, sep_value) VALUES ($1, $2, $3) returning cid"
+    }
+}
+
+#[derive(Debug, Clone, sqlx::FromRow)]
+pub struct StreamCommitRow {
+    pub cid: Vec<u8>,
+    pub prev: Option<Vec<u8>>,
+    pub delivered: bool,
+}
+
+impl StreamCommitRow {
+    pub fn fetch_by_stream_cid() -> &'static str {
+        r#"
+        SELECT eh.cid as "cid", eh.prev as "prev", 
+            e.delivered IS NOT NULL as "delivered" 
+        FROM ceramic_one_stream s
+            JOIN ceramic_one_event_header eh on eh.stream_cid = s.cid
+            JOIN ceramic_one_event e on e.cid = eh.cid
+        WHERE s.cid = $1"#
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct IncompleteStream {
```

Comment:
comment on the type please.  This PR introduces a lot of new types and I'm worried about how confusing it will be to keep track of them all and how they're different.

I believe this type represents a stream where we know about some events but there are holes in the history meaning some of the events are still undeliverable - is that right?  Can we add a comment to that effect?

Reply:
yeah, I will review/consolidate the types. This is basically a wrapper around a CID for serialization to support a specific query and is only public to the module.